diff --git a/src/EntityReferenceRevisionsServiceProvider.php b/src/EntityReferenceRevisionsServiceProvider.php
index 65db785..b5d3310 100644
--- a/src/EntityReferenceRevisionsServiceProvider.php
+++ b/src/EntityReferenceRevisionsServiceProvider.php
@@ -21,7 +21,7 @@ class EntityReferenceRevisionsServiceProvider extends ServiceProviderBase {
       // Hal module is enabled, add our new normalizer for entity reference
       // revision items.
       $service_definition = new Definition('Drupal\entity_reference_revisions\Normalizer\EntityReferenceRevisionItemNormalizer', array(
-        new Reference('hal.link_manager'),
+        new Reference('rest.link_manager'),
         new Reference('serializer.entity_resolver'),
       ));
       // The priority must be higher than that of
