diff -upr vendor/solarium/solarium/library/Solarium/Core/Client/Adapter/Http.php /Users/balu/solarium-backup/solarium/library/Solarium/Core/Client/Adapter/Http.php
--- vendor/solarium/solarium/library/Solarium/Core/Client/Adapter/Http.php	2015-12-19 17:00:15.000000000 +0100
+++ /Users/balu/solarium-backup/solarium/library/Solarium/Core/Client/Adapter/Http.php	2016-03-31 11:14:25.000000000 +0200
@@ -111,13 +111,25 @@ class Http extends Configurable implemen
 
         if ($method == Request::METHOD_POST) {
             if ($request->getFileUpload()) {
+                $boundary = '----------' . md5(time());
+                $CRLF = "\r\n";
+                $file = $request->getFileUpload();
+                $filename = basename($file);
+                // Add the proper boundary to the Content-Type header
+                $request->addHeader("Content-Type: multipart/form-data; boundary={$boundary}");
+                $data =  "--{$boundary}" . $CRLF;
+                $data .= 'Content-Disposition: form-data; name="upload"; filename=' . $filename . $CRLF;
+                $data .= 'Content-Type: application/octet-stream' . $CRLF . $CRLF;
+                $data .= file_get_contents($request->getFileUpload()) . $CRLF;
+                $data .= '--' . $boundary . '--';
+                $content_length = strlen($data);
+                $request->addHeader("Content-Length: $content_length\r\n");
                 stream_context_set_option(
                     $context,
                     'http',
                     'content',
-                    file_get_contents($request->getFileUpload())
+                    $data
                 );
-                $request->addHeader('Content-Type: multipart/form-data');
             } else {
                 $data = $request->getRawData();
                 if (null !== $data) {
diff -upr vendor/solarium/solarium/library/Solarium/QueryType/Extract/RequestBuilder.php /Users/balu/solarium-backup/solarium/library/Solarium/QueryType/Extract/RequestBuilder.php
--- vendor/solarium/solarium/library/Solarium/QueryType/Extract/RequestBuilder.php	2015-12-19 17:00:15.000000000 +0100
+++ /Users/balu/solarium-backup/solarium/library/Solarium/QueryType/Extract/RequestBuilder.php	2016-03-31 11:14:25.000000000 +0200
@@ -106,7 +106,6 @@ class RequestBuilder extends BaseRequest
         } elseif (is_readable($file)) {
             $request->setFileUpload($file);
             $request->addParam('resource.name', basename($query->getFile()));
-            $request->addHeader('Content-Type: multipart/form-data');
         } else {
             throw new RuntimeException('Extract query file path/url invalid or not available');
         }
